aeneas/src/ir/Normalization.v3

```diff
@@ -421,18 +334,35 @@ class ReachabilityNormalizer(config: NormalizerConfig, ra: ReachabilityAnalyzer)
 	}
 	// normalize a live instance of a class
 	def normClassRecord(rc: RaClass, oldRecord: Record, array: Array<Val>) {
-		if (rc.dataNorm != null && rc.dataNorm.tagType != null) {
-			array[0] = Int.box(V3.getVariantTag(oldRecord.rtype));
+		var vrc = ra.getClass(oldRecord.rtype), vn = vrc.variantNorm;
+		if (vn == null) {
+			var rfs = rc.fields;
+			for (i < rfs.length) {
+				var rf = rfs[i];
+				if (rf != null && rf.normIndex >= 0) {
+					var v = oldRecord.values[i];
+					if (rf.fieldType == null) array[rf.normIndex] = normSimpleVal(null, v);
+					else normValIntoArray(v, rf.typeNorm, array, rf.normIndex);
+				}
+			}
+			return;
 		}
-		var rfs = rc.fields;
-		for (i < rfs.length) {
-			var rf = rfs[i];
-			if (rf != null && rf.normIndex >= 0) {
-				var v = oldRecord.values[i];
-				if (rf.fieldType == null) array[rf.normIndex] = normSimpleVal(null, v);
-				else normValIntoArray(v, rf.typeNorm, array, rf.normIndex);
+
+		if (!vn.isEnum()) {
```

Comment:
What about returning early if it's an enum? (Thus the following code is indented less.